lib/sinon/test.js

```diff
@@ -55,6 +55,23 @@
                     sandbox.verifyAndRestore();
                 }
 
+                if (result) {
+                    if (typeof Promise === "function") {
```

Comment:
don't see the point in this check. we don't really care if the global environment supports Promises: we care about the actual returned object. So just use the thenable check on line 62 and drop this. This could fail too:
Say, a promise was returned that was not native (pinkySwear, for instance). Then this test would say `isPromise == false`. That's wrong. So just focus on if it is a Thenable or not.